### PR TITLE
Refine replace_blob method to raise error while deleting missing blob,

### PR DESCRIPTION
**Pull request by xingzhou**
rather than doing extra existence check.

[#101003512](https://www.pivotaltracker.com/story/show/101003512)

Signed-off-by: Xing Zhou <xingzhou@cn.ibm.com>

### CODE REVIEW

File: bosh-director/lib/bosh/director/blob_util.rb [hunk 1]
```diff
@@ -23,7 +23,11 @@ def self.copy_blob(blobstore_id)
     end
 
     def self.replace_blob(old_blobstore_id, path)
-      blobstore.delete(old_blobstore_id) if blobstore.exists?(old_blobstore_id)
+      begin
+        blobstore.delete(old_blobstore_id)
+      rescue Bosh::Blobstore::BlobstoreError => e
```
Comment (cppforlife) [inline bosh-director/lib/bosh/director/blob_util.rb]: oops forgot to mention. let's make sure to log to debug about failure ignore. i believe we do similar thing in other places.

File: bosh-director/lib/bosh/director/jobs/update_release.rb [hunk 2]
```diff
@@ -638,10 +635,15 @@ def next_release_version
...
 
       def fix_package(package, package_tgz)
-        single_step_stage("Fixing package '#{package.name}/#{package.version}'") do
-          logger.info("Fixing package '#{package.name}/#{package.version}'")
-          package.blobstore_id = BlobUtil.replace_blob(package.blobstore_id, package_tgz)
+        begin
+          logger.info("Deleting package '#{package.name}/#{package.version}'")
+          BlobUtil.delete_blob(package.blobstore_id)
+        rescue Bosh::Blobstore::BlobstoreError => e
+          logger.info("Error deleting '#{package.name}/#{package.version}' #{e.message}")
```
Comment (cppforlife) [inline bosh-director/lib/bosh/director/jobs/update_release.rb]: Please include blobstore_id and e.inspect in the error log.

**Comment by guoger** [comment]
@cppforlife , just to comment on your rejection in the tracker, package.sha1 should not be affected since we are replacing corrupted blobs with correct ones, which should already match the sha1 in database, since they are identical packages.

Thanks!
/Jay

**Comment by cppforlife** [comment]
@guoger they may be identical in content (fingerprint is the same) but because tar generates different file for the same content sha1 may not match.

**Comment by guoger** [comment]
@cppforlife That's true, unlike simple tar, gzip (-z) indeed produces different bits out of same content. Didn't know that and thanks for pointing it out.

**Pull request merged**